ui/src/components/UploadStory.tsx

```diff
@@ -0,0 +1,855 @@
+    currentStory.author_country
+  );
+  const [currentCity, setCurrentCity] = useState(
+    currentStory.current_city.toUpperCase()
+  );
+  const [autocompleteAuthorCountry, setAutocompleteAuthorCountry] = useState(
+    currentStory.author_country
+  );
+  const [addedCountry, setAddedCountry] = useState("");
+  const [newImage, setNewImage] = useState(currentStory.image_url);
+  const [disabled, setDisabled] = useState(false);
+  //handleSubmit component states
+  const [dialogOpenState, setDialogOpenState] = useState(false);
+  const [uploadErrorState, setErrorOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+
+  const { data: countries, error: errorCountries } = useSWR<string[]>(
+    "/api/countries"
+  );
+
+  const yearArray = [];
+  let year: number = new Date().getFullYear();
+  for (let i = 30; i > 0; i--) {
+    yearArray.push(year);
+    year -= 1;
+  }
+
+  const [formInput, setFormInput] = useReducer(
+    (state, newState) => ({ ...state, ...newState }),
+    {
+      image: new File([""], "filename"),
+      video_url: currentStory.video_url,
+      title: currentStory.title,
+      content: currentStory.content,
+      summary: currentStory.summary,
+      author_first_name: currentStory.author_first_name,
+      author_last_name: currentStory.author_last_name,
+      author_country: currentStory.author_country,
+      year: currentStory.year,
+      current_city: currentStory.current_city.toUpperCase(),
+      bio: bio,
+    }
+  );
+
+  const story = React.useMemo(() => {
+    if (isDrawerOpen) {
+      const storyFromData: Story = {
+        image_url:
+          newImage == ""
+            ? URL.createObjectURL(formInput.image)
+            : currentStory.image_url,
+        video_url: formInput.video_url as string,
+        title: formInput.title as string,
+        content: formInput.content as string,
+        summary: formInput.summary as string,
+        author_first_name: formInput.author_first_name as string,
+        author_last_name: formInput.author_last_name as string,
+        author_country: authorCountry as string,
+        year: formInput.year as number,
+        current_city: formInput.current_city as string,
+        author: {
+          ID: 0,
+          CreatedAt: "",
+          UpdatedAt: "",
+          DeletedAt: {
+            Time: "",
+            Valid: true,
+          },
+          firstName: formInput.author_last_name as string,
+          currentCity: formInput.current_city as string,
+          bio: formInput.bio as string,
+        } as Author,
+        tags: [],
+        is_visible: true,
+        latitude: 0,
+        longitude: 0,
+        ID: 0,
+        CreatedAt: "",
+        UpdatedAt: "",
+        DeletedAt: {
+          Time: "",
+          Valid: true,
+        },
+      };
+      return storyFromData;
+    }
+    return undefined;
+  }, [isDrawerOpen, formInput]);
+
```

Comment:
@hanlinc27 @AbhiPrasad @aspensantos is it okay to leave this warning alone ^^? The code is from Aspen's commit so I'm not too sure what the problem is